File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) 
    {
        HashSet<char> substring = new HashSet<char>();
        int max = 0;
        int i = 0;
        int j = 0; 
        while(j < s.Length)
        {
            char front = s[j];
            char back = s[i];
            if(substring.Contains(front))
            {
                substring.Remove(back);
                i++;
            }
            else
            {
                substring.Add(front);
                j++;
            }
            if(substring.Count > max)
            {
                max = substring.Count;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) 
    {
        HashSet<Integer> numbers = new HashSet<Integer>();
        int maxcount = 0;
        for(int i = 0; i < nums.length; i++)
        {
            numbers.add(nums[i]);
        }
        for(Integer num : numbers)
        {
            if(!numbers.contains(num - 1))
            {
                Integer newnum = num - 1;
                int count = 0;
                while(numbers.contains(newnum + 1))
                {
                    newnum = newnum + 1;
                    count = count + 1;
                }
                maxcount = Math.max(count, maxcount);
                
            }
        }
        return maxcount;
    }
}
